File: frameworks/node/koa/UnitTests/json.test.ts
import assert from "node:assert/strict";
import { test } from "node:test";

import { client, expected } from "./app.ts";

// rb:test json.small,json.medium,json.large
for (const [id, size] of [["json.small", "small"], ["json.medium", "medium"], ["json.large", "large"]] as const) {
  test(`${id}: the payload is written with JSON.stringify`, async () => {
    const response = await client.get(`/json/${size}`);

    assert.equal(response.status, 200);
    assert.match(response.headers["content-type"]!, /^application\/json/);
    assert.deepEqual(response.body, expected.json(`items.${size}.json`));
  });
}

File: frameworks/go/huma/UnitTests/json_test.go
package unittests

import "testing"

// rb:test json.small,json.medium,json.large
func TestJSON(t *testing.T) {
	for _, row := range []struct{ id, path, file string }{
		{"json.small", "/json/small", "items.small.json"},
		{"json.medium", "/json/medium", "items.medium.json"},
		{"json.large", "/json/large", "items.large.json"},
	} {
		t.Run(row.id, func(t *testing.T) {
			response := client(t).Get(row.path)

			assertOK(t, payload(t, row.file), response)
			assertHeader(t, response, "Content-Type", "application/json")
		})
	}
}
